trantor/net/inner/TcpConnectionImpl.cc

```diff
@@ -1295,16 +1304,22 @@ void TcpConnectionImpl::sendFile(const char *fileName,
     }
 
     sendFile(fd, offset, length);
-#else
-#ifndef _MSC_VER
-    auto fp = fopen(fileName, "rb");
-#else
+#endif  // _WIN32
+}
+
+#ifdef _WIN32
+void TcpConnectionImpl::sendFile(const wchar_t *fileName,
+                                 size_t offset,
+                                 size_t length)
+{
+    assert(fileName);
     FILE *fp;
-    if (fopen_s(&fp, fileName, "rb") != 0)
-    {
+#ifndef _MSC_VER
+    fp = _wfopen(fileName, L"rb");
+#else   // _MSC_VER
+    if (_wfopen_s(&fp, fileName, L"rb") != 0)
```

Comment:
I'm not sure how [clang for Windows](https://docs.microsoft.com/en-us/cpp/build/clang-support-msbuild?view=msvc-160) would react to this. Should work both ways. Please try it if it's not a big hurdle.